components/kyma-environment-broker/internal/avs/delegator.go

```diff
@@ -0,0 +1,117 @@
+package avs
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/process"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/storage"
+	"github.com/kyma-incubator/compass/components/provisioner/pkg/gqlschema"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	overrideKey     = "avs_bridge.config.evaluations.cluster.id"
+	avsBridgeAPIKey = "avs_bridge.config.availabilityService.apiKey"
+)
+
+type delegator struct {
+	operationManager *process.OperationManager
+	avsConfig        Config
+	clientHolder     *clientHolder
+}
+
+func newDelegator(avsConfig Config, operationsStorage storage.Operations) *delegator {
+	return &delegator{
+		operationManager: process.NewOperationManager(operationsStorage),
+		avsConfig:        avsConfig,
+		clientHolder:     newClientHolder(avsConfig),
+	}
+}
+
+func (del *delegator) doRun(logger logrus.FieldLogger, stepName string, operation internal.ProvisioningOperation,
+	modelSupplier func(provisioningOperation internal.ProvisioningOperation) (*basicEvaluationCreateRequest, error)) (internal.ProvisioningOperation, time.Duration, error) {
+	logger.Infof("starting the step %s", stepName)
```

Comment:
The logger contains information about operation id, instance id and the step name: https://github.com/kyma-incubator/compass/blob/master/components/kyma-environment-broker/internal/process/manager.go#L53   and https://github.com/kyma-incubator/compass/blob/master/components/kyma-environment-broker/internal/process/manager.go#L59